lib/source_file.cpp

```diff
@@ -226,6 +226,36 @@ cstring InputSources::getSourceFragment(const SourceInfo &position) const {
     return result + toadd + marker + cstring::newline;
 }
 
+cstring InputSources::getSourceFragment2(const SourceInfo &position) const {
```

Comment:
Agreed for sure on the name.  It was a placeholder name when I wrote it.

Do you have a pointer to a Doxygen comment I could use as an example to pattern a new one after?  source_file.cpp and source_file.h don't seem to have any right now.